presto-main/src/test/java/com/facebook/presto/operator/PageAssertions.java

```diff
@@ -36,4 +48,96 @@ public static void assertPageEquals(List<? extends Type> types, Page actualPage,
+    }
+
+    public static Page createDictionaryPage(List<Type> types, int positionCount, boolean allowNulls)
+    {
+        return createPage(types, positionCount, true, false, allowNulls, false, ImmutableList.of(DICTIONARY));
+    }
+
+    public static Page createRLEPage(List<Type> types, int positionCount, boolean allowNulls)
+    {
+        return createPage(types, positionCount, true, false, allowNulls, false, ImmutableList.of(RUN_LENGTH));
+    }
+
+    public static Page createPage(List<Type> types, int positionCount, boolean addPreComputedHashBlockAtBegin, boolean addNullBlockAtEnd, boolean useBlockView, List<Encoding> wrappings)
+    {
+        return createPage(types, positionCount, addPreComputedHashBlockAtBegin, addNullBlockAtEnd, true, useBlockView, wrappings);
+    }
+
+    private static Page createPage(
+            List<Type> types,
+            int positionCount,
+            boolean addPreComputedHashBlock,
+            boolean addNullBlock,
+            boolean allowNulls,
+            boolean useBlockView,
+            List<Encoding> wrappings)
+    {
+        int channelCount = types.size();
+        int preComputedChannelCount = (addPreComputedHashBlock ? 1 : 0);
+        int nullChannelCount = (addNullBlock ? 1 : 0);
+
+        Block[] blocks = new Block[channelCount + preComputedChannelCount + nullChannelCount];
+
+        if (addPreComputedHashBlock) {
+            blocks[0] = createLongsBlock(positionCount, false);
+        }
+
+        for (int i = 0; i < channelCount; i++) {
+            blocks[i + preComputedChannelCount] = createBlockWithType(types.get(i), positionCount, allowNulls, useBlockView, wrappings);
+        }
+
+        if (addNullBlock) {
+            blocks[channelCount + preComputedChannelCount] = createNullBlock(BIGINT, positionCount);
+        }
+
+        return new Page(positionCount, blocks);
+    }
+
+    public static Page mergePages(List<Type> types, List<Page> pages)
+    {
+        PageBuilder pageBuilder = new PageBuilder(types);
+        int totalPositionCount = 0;
+        for (Page page : pages) {
+            verify(page.getChannelCount() == types.size());
+
+            for (int i = 0; i < types.size(); i++) {
+                BlockBuilder blockBuilder = pageBuilder.getBlockBuilder(i);
+                Block block = page.getBlock(i);
+                for (int position = 0; position < page.getPositionCount(); position++) {
+                    if (block.isNull(position)) {
+                        blockBuilder.appendNull();
+                    }
+                    else {
+                        block.writePositionTo(position, blockBuilder);
+                    }
+                }
+            }
+            totalPositionCount += page.getPositionCount();
+        }
+        pageBuilder.declarePositions(totalPositionCount);
+        return pageBuilder.build();
+    }
+
+    public static List<Type> buildTypes(List<Type> types, boolean addPreComputedHashBlock, boolean addNullBlock)
```

Comment:
@arhimondr I added this method because it's looks confusing by always constructing the target types in the tests since they are not our main target of testing. There're multiple places using this method. How about adding some comments and changing the name to the following?

```
    /**
     * Create a new types list that prepends the BIGINT type in front if addPreComputedHashBlock is true, and appends the BIGINT type at the end if addNullBlock is true.
     */
    public static List<Type> updateBlockTypesWithHashBlockAndNullBlock(List<Type> types, boolean addPreComputedHashBlock, boolean addNullBlock)
```